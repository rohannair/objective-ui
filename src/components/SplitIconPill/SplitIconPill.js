import React, { PropTypes } from 'react'
import styled from 'styled-components'
import Pill from '../Pill'

const SplitIconPill = p => (
  <div className='spliticonpill'>
    <Pill
      { ...p }
      transparent
      onClick={
        p.iconAction
        ? p.iconAction
        : p.action
      }>
      <i className={`zmdi zmdi-${p.icon}`}/>
    </Pill>
    <Pill {...p} onClick={p.action}>{ p.children }</Pill>
  </div>
)

export default SplitIconPill
