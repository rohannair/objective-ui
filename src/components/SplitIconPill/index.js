import SplitIconPill from './SplitIconPill'
export default SplitIconPill
