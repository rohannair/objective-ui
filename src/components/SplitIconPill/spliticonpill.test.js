import React from 'react'

import SplitIconPill from './index.js'

describe('Test suite for SplitIconPill component', () => {
  it('SplitIconPill should exist', () => {
    let wrapper = shallow(<SplitIconPill/>)
    expect(wrapper).toMatchSnapshot()
  })
})

