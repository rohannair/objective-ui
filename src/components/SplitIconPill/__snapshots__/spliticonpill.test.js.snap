exports[`Test suite for SplitIconPill component SplitIconPill should exist 1`] = `
<div
  className="spliticonpill">
  <Pill
    transparent={true}>
    <i
      className="zmdi zmdi-undefined" />
  </Pill>
  <Pill />
</div>
`;
